### PR TITLE
feat: [UIE-8137] - IAM RBAC: add new user details component

### PR DESCRIPTION
**Pull request by aaleksee-akamai**
## Description 📝

IAM RBAC - user details component.
This component is almost identical to an existing one in the `Account`, with some adjustments.
One difference is the addition of a new field showing the number of assigned roles.

## Changes  🔄

List any change(s) relevant to the reviewer.

- new user details component

## Target release date 🗓️

1/14/25 (dev)

## Preview 📷

**Include a screenshot or screen recording of the change.**

:lock: Use the [Mask Sensitive Data](https://cloud.linode.com/profile/settings) setting for security.

:bulb: Use `<video src="" />` tag when including recordings in table.

| Before  | After   |
| ------- | ------- |
| 📷 |image|
| 📷 |image|

### Prerequisites

- Ensure the Identity and Access Beta flag is enabled in dev tools
- Click on the username in the users table or click on the user's menu `View User Details`

### Verification steps

(How to verify changes)

- [ ] Confirm user details renders.

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>

- [x] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [x] All unit tests are passing
- [x] TypeScript compilation succeeded without errors
- [x] Code passes all linting rules

</details>

### CODE REVIEW

**Review by cpathipa — COMMENTED** [review]
Missing changeset for API-V4

**Comment by hkhalil-akamai** [comment]
Will the IAM user details page replace the exiting Account/Users page?

If not, it seems like these components share significant component code with the existing components from *src/features/Users/UserProfile* -- can we find a way to abstract the components so we can avoid duplicating the code?

@aaleksee-akamai

**Comment by aaleksee-akamai** [comment]
> Will the IAM user details page replace the exiting Account/Users page?
> 
> If not, it seems like these components share significant component code with the existing components from _src/features/Users/UserProfile_ -- can we find a way to abstract the components so we can avoid duplicating the code?
> 
> @aaleksee-akamai

yes, it will

@hkhalil-akamai

File: packages/manager/src/features/IAM/Users/UserDetails/DeleteUserPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import { Box, Button, Paper, Stack, Typography } from '@linode/ui';
```
Comment (cpathipa) [inline packages/manager/src/features/IAM/Users/UserDetails/DeleteUserPanel.tsx:1]: Can we have unit test coverage for this?

**Review by cpathipa — APPROVED** [review]
Overall LGTM! confirming on the verification steps..

UXObservation - 

1. I feel that there is a lot of white space on the user details page, where each card contains a single field with a button. I think we can revisit the page's UX and simplify it by consolidating the fields and using a single save button.

image

**Comment by cpathipa** [comment]
@aaleksee-akamai Conflicts should be resolved and updated with latest before merging this branch.

File: packages/manager/src/features/IAM/Users/UserDetails/DeleteUserPanel.tsx (hunk 1 above)
Comment (aaleksee-akamai) [reply packages/manager/src/features/IAM/Users/UserDetails/DeleteUserPanel.tsx:1]: thanks @cpathipa , done

**Review by hkhalil-akamai — APPROVED** [review]

**Pull request merged**